hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1192,15 +1192,14 @@ int QuectelNcpClient::registerNet() {
     connectionState(NcpConnectionState::CONNECTING);
 
     auto resp = parser_.sendCommand("AT+COPS?");
-    int copsState = -1;
+    int copsState = 2;
     r = CHECK_PARSER(resp.scanf("+COPS: %d", &copsState));
     CHECK_TRUE(r == 1, SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED);
     r = CHECK_PARSER(resp.readResult());
     CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_AT_NOT_OK);
 
-    if (copsState != 0) {
-        // Only run AT+COPS=0 if not currently registering, otherwise we will
-        // perform PLMN reselection
+    if (copsState == 2) {
```

Comment:
Should we instead use `copsState !=0 && copsState != 1` to also handle potentially cases where `AT+COPS?` may return something other than 0, 1 or 2?

Reply:
![image](https://user-images.githubusercontent.com/48732259/88558077-b2f93a80-cff0-11ea-9774-7f7249cc1ae2.png)

I think it should be fine? At least for mode `4`, I think it's ok, but idk what mode `3` does.

Reply:
Mode 3 just sets the format (numeric/alphanumeric).  We actually set this to numeric for MCCMNC reporting.  It believe it should never return +COPS: 3.

Reply:
Well, there is 4 as well.

Reply:
Actually may be good to use `copsState !=0 && copsState != 1` justtt in case we get "bad" values from COPS?

Reply:
`copsState != 0 && copsState != 1 && copsState != 3 && copsState != 4` then? :D

Reply:
0 and 1 should probably be enough, right? Because we want to avoid PLMN reselection only in those cases. Anything else - fair game.

Reply:
I removed 3... because if for some odd reason it throws back a 3, well then I don't trust it... and we can set COPS=0.

Reply:
I'm mainly offering this because the appnote that we've kind of based this on from ublox clearly specifies that the action needs to be taken on COPS != 0, so if we are adding something to support 1, IMO we should specifically target that.

Reply:
Updated to `copsState != 0 && copsState != 1` as discussed.